### PR TITLE
Creation of UI for login feature

### PR DESCRIPTION
**Pull request by rickwolox**
### Summary
User interface creation for Login Feature in Training App.

### Screenshot
Screenshot from 2019-07-19 16-46-44

### Trello Card
https://trello.com/c/oz3DxsQs/16-u1-frontend-login-componentes-b%C3%A1sicos

### CODE REVIEW

File: app/src/main/res/drawable/button_blue.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<shape
+    xmlns:android="http://schemas.android.com/apk/res/android">
+    <stroke
+        android:width="1dp"
```
Comment (lmiotti) [inline app/src/main/res/drawable/button_blue.xml]: Solve harcoded dimens and colors! There are files like dimens.xml and colors.xml where you can save them (there are some already saved)

File: app/src/main/res/drawable/gradient_background_top.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle" >
+
+    <gradient
+        android:angle="90"
+        android:endColor="@color/colorPrimary"
+        android:startColor="@color/colorAccent" />
+
+    <corners android:radius="0dp" />
```
Comment (lmiotti) [inline app/src/main/res/drawable/gradient_background_top.xml]: Same as above

File: app/src/main/res/layout-v21/fragment_login.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
```
Comment (lmiotti) [inline app/src/main/res/layout-v21/fragment_login.xml]: Remove this empty line

File: app/src/main/res/layout-v21/fragment_login.xml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    <ImageView
+        android:id="@+id/vBackgroundTop"
```
Comment (lmiotti) [inline app/src/main/res/layout-v21/fragment_login.xml]: We are going to order properties like this: android -> app -> tools -> style. Change themm to suite this format. Moreover, add a more descriptive name, like vLoginBackgroundTop.

File: app/src/main/res/layout-v21/fragment_login.xml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="@id/vBackgroundTop"
+        android:alpha="0.8"
+        android:background="@drawable/gradient_background_top" />
+
+    <ImageView
+        android:id="@+id/vLogoLarge"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
```
Comment (lmiotti) [inline app/src/main/res/layout-v21/fragment_login.xml]: Solve these harcoded dimens!

File: app/src/main/res/layout-v21/fragment_login.xml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        android:orientation="vertical"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.6"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/vLogoLarge">
+
+        <EditText
+            android:id="@+id/vUsername"
+            style="@style/styleBigSpacing"
+            android:ems="10"
```
Comment (lmiotti) [inline app/src/main/res/layout-v21/fragment_login.xml]: Which is the meaning of this magic number? We can move it to an integer file!

File: app/src/main/res/layout-v21/fragment_login.xml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            android:hint="@string/login_username"
+            android:inputType="textPersonName"
+            tools:ignore="Autofill" />
+
+        <EditText
+            android:id="@+id/vPassword"
```
Comment (lmiotti) [inline app/src/main/res/layout-v21/fragment_login.xml]: Use a more descriptive name, such as vLoginPassword

File: app/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -8,4 +8,16 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+
+    <!-- Layout Styles -->
+    <style name="styleBigSpacing">
+        <item name="android:layout_height">56dp</item>
```
Comment (lmiotti) [inline app/src/main/res/values/styles.xml]: Solve these harcoded dimens

File: app/src/main/res/layout/fragment_login.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_login.xml:1]: Same comments as fragment_login in v21

**Review by lmiotti — CHANGES_REQUESTED** [review]
Please, check the corrections I've marked and tell me if you need help. Keep on working like this!

File: app/src/main/res/layout-v21/fragment_login.xml (hunk 6 above)
Comment (rickwolox) [reply app/src/main/res/layout-v21/fragment_login.xml]: that number represents the width of the font. It was set by default when I drag and drop the element to the layout.

**Review by lmiotti — APPROVED** [review]
First PR is approved! Good job!

james_corden

**Pull request merged**